patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/SampleLoader.java

```diff
@@ -53,7 +53,35 @@ public SampleLoader(@NotNull final Lims lims) {
     }
 
     @NotNull
-    private static List<RunContext> loadRunContexts(@NotNull String runsDirectory, @NotNull String pipelineVersionFile) throws IOException {
+    private List<RunContext> extractRunContextsFromDirectoryOrJson(final @NotNull ClinicalAlgoConfig config, final String runsDirectory,
+            final String runsJson) throws IOException {
+        List<RunContext> runContexts;
+        if (runsDirectory != null) {
+            String pipelineVersionFile = config.pipelineVersionFile();
```

Comment:
Given this is a new addition to an existing command line, should we consider also adding logging if both `runsDirectory` and `runsJson` are given (to tell the user that the JSON will be ignored essentially)?

Reply:
yes lets throw when neither are specified or both